arangod/Replication2/ReplicatedLog/LogLeader.cpp

```diff
@@ -1250,20 +1260,86 @@ auto replicated_log::LogLeader::waitForLeadership()
 }
 
 auto replicated_log::LogLeader::updateParticipantsConfig(
-    std::shared_ptr<ParticipantsConfig const> config) -> LogIndex {
+    std::shared_ptr<ParticipantsConfig const> config,
+    std::size_t previousGeneration,
+    std::unordered_map<ParticipantId, std::shared_ptr<AbstractFollower>>
+        additionalFollowers,
+    std::vector<ParticipantId> const& followersToRemove) -> LogIndex {
   LOG_CTX("ac277", DEBUG, _logContext)
       << "updating configuration to generation " << config->generation;
+  TRI_ASSERT(previousGeneration < config->generation);
   auto waitForIndex = _guardedLeaderData.doUnderLock([&](GuardedLeaderData&
                                                              data) {
-    if (data.activeParticipantConfig->generation >= config->generation) {
+    if (data.activeParticipantsConfig->generation >= config->generation) {
       THROW_ARANGO_EXCEPTION_FORMAT(
           TRI_ERROR_BAD_PARAMETER,
           "updated participant config generation is smaller or equal to "
           "current generation - refusing to update; new = %zu, current = %zu",
-          config->generation, data.activeParticipantConfig->generation);
+          config->generation, data.activeParticipantsConfig->generation);
+    }
+    if (data.activeParticipantsConfig->generation != previousGeneration) {
+      // This is to make sure the `additionalFollowers` list is really the
+      // (asymmetric) difference between the current and new configuration.
+      THROW_ARANGO_EXCEPTION_FORMAT(
+          TRI_ERROR_BAD_PARAMETER,
+          "assumed participant config generation does not match the current "
+          "generation - refusing to update; previous = %zu, current = %zu",
+          previousGeneration, data.activeParticipantsConfig->generation);
+    }
```

Comment:
If there is not test for this, we should make sure to add one. This small piece of code is very important! 🙂

Reply:
Agreed, I'll add tests.